src/main/java/io/bartek/cat/CalculatorStart.java

```diff
@@ -1,22 +1,67 @@
-package io.bartek.cat;public class CalculatorStart {
+package io.bartek.cat;
+
+import java.sql.Array;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CalculatorStart {
     private final CalculatorDecisionHandler calculatorDecisionHandler;
     private final UserInterface userInterface;
+    private final FileInputHandler fileInputHandler;
 
-    public CalculatorStart(CalculatorDecisionHandler calculatorDecisionHandler, UserInterface userInterface) {
+    public CalculatorStart(CalculatorDecisionHandler calculatorDecisionHandler, UserInterface userInterface, FileInputHandler fileInputHandler) {
         this.calculatorDecisionHandler = calculatorDecisionHandler;
         this.userInterface = userInterface;
+        this.fileInputHandler = fileInputHandler;
     }
+
+
     public void startCalculator() {
+        userInterface.printMessage("Type 'manual' if you want to input data manually or 'file' if you want to load it from file ");
+        boolean inputTypeisValid = false;
+
+        while (!inputTypeisValid) {
+            String inputType = userInterface.getUserInput().trim();
+            switch (inputType) {
+                case "manual" -> {
+                    handleManualPart();
+                    inputTypeisValid = true;
+                }
+                case "file" -> {
+                    handleFilePart();
+                    inputTypeisValid = true;
+                }
+                default -> userInterface.printMessage("Please type 'manual' or 'file'.");
+            }
+        }
+    }
+
+    private void handleFilePart() {
+        userInterface.printMessage("Provide the file path");
+        String filePath = userInterface.getUserInput();
+        try {
+            String fileData = fileInputHandler.readFile(filePath);
+            userInterface.printMessage("Data from file: " + fileData);
+
+            int result = calculatorDecisionHandler.handleFileData(fileData);
```

Comment:
I think now responsibility of calculatorDecisionHandler is mixed (Has 2 reason to change changes in file and console input). Please provide common format for both types of input and execute calculations.

Reply:
Yup I’m aware of that, it need to be done in next step 🫡

Reply:
Ok, please continue your task